plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -240,21 +240,23 @@ export class EventsProcessor {
         }
 
         const useExternalSchemas = this.clickhouseExternalSchemasEnabled(teamId)
-        // proto ingestion is deprecated and we won't support new additions to the schema
-        const message = useExternalSchemas
-            ? (EventProto.encodeDelimited(EventProto.create(eventPayload)).finish() as Buffer)
-            : Buffer.from(
-                  JSON.stringify({
-                      ...eventPayload,
-                      person_id: personInfo?.uuid,
-                      person_properties: eventPersonProperties,
-                      person_created_at: personInfo
-                          ? castTimestampOrNow(personInfo?.created_at, TimestampFormat.ClickHouseSecondPrecision)
-                          : null,
-                      ...groupsProperties,
-                      ...groupsCreatedAt,
-                  })
-              )
+        let message: Buffer
+        if (useExternalSchemas) {
+            // Proto ingestion, which is deprecated - we won't support new additions to the schema
+            message = EventProto.encodeDelimited(EventProto.create(eventPayload as IEvent)).finish() as Buffer
+        } else {
+            const rawEvent: RawEvent = {
+                ...eventPayload,
+                person_id: personInfo?.uuid,
+                person_properties: eventPersonProperties,
+                person_created_at: personInfo
+                    ? castTimestampOrNow(personInfo?.created_at, TimestampFormat.ClickHouseSecondPrecision)
+                    : undefined,
```

Comment:
Why  change null to undefined here? They're not equivelent.

Reply:
Not equivalent, but this more accurately reflects the typing (previously this was a loose untyped object, now it's ensured to be `RawEvent`). It doesn't _seem_ to have affected anything, looking at E2E ingestion tests.